internal/api/update_credentials_v1alpha1.go

```diff
@@ -90,6 +92,15 @@ func applyCredentialsUpdateToSecret(
 	secret *corev1.Secret,
 	credsUpdate credentialsUpdate,
 ) {
+	if credsUpdate.description != "" {
+		if secret.Annotations == nil {
+			secret.Annotations = make(map[string]string)
```

Comment:
Nit: You can initialize this with a capacity of 1 to avoid an resizing when the annotation is added to it.

This is non-blocking.